opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -27,6 +25,27 @@ impl<'a> tracing::field::Visit for EventVisitor<'a> {
         }
     }
 
+    fn record_str(&mut self, field: &tracing_core::Field, value: &str) {
+        if let Some(ref mut map) = self.log_record.attributes {
+            map.insert(field.name().into(), value.to_owned().into());
+        } else {
+            let mut map = OrderMap::with_capacity(1);
+            let k: Key = field.name().into();
```

Comment:
Style nit: confused why you bind `k` and `v` here rather than inlining them as you've done in `record_bool()`. The type annotations should not be necessary here.

Reply:
I got the error without type annotations but then it works after specifying the explicit type for OrderMap. Have done the changes. Thanks for the comment.